### PR TITLE
chg:usr:Add a CHANGELOG File

### PR DESCRIPTION
**Pull request by vrajat**
Create a new file to hold CHANGELOG. The changelog is generated using [gitchangelog](https://github.com/vaab/gitchangelog). The required RC file has been added in this commit
as well. Instructions regarding commit message formats and to generate the CHANGELOG will
be added in a subsequent commit.

### CODE REVIEW

**Comment by vrajat** [comment]
@GodenYao Can you please review this PR ?

File: .gitchangelog.rc [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+## ``section_regexps`` is a list of 2-tuples associating a string label and a
+## list of regexp
+##
+## Commit messages will be classified in sections thanks to this. Section
+## titles are the label, and a commit is classified under this section if any
+## of the regexps associated is matching.
+##
+## Please note that ``section_regexps`` will only classify commits and won't
+## make any changes to the contents. So you'll probably want to go check
```
Comment (GodenYao) [inline .gitchangelog.rc]: content

File: .gitchangelog.rc [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        r'^[fF]ix\s*:\s*((dev|use?r|pkg|test|doc)\s*:\s*)?([^\n]*)$',
+     ]),
+
+    ('Other', None ## Match all lines
+     ),
+
+]
+
+
+## ``body_process`` is a callable
```
Comment (GodenYao) [inline .gitchangelog.rc]: what's "callable"?

File: .gitchangelog.rc [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+## be used in the changelog.
+##
+## Available constructs are:
+##
+##   - any python callable that take one txt argument and return txt argument.
+##
+##   - ReSub(pattern, replacement): will apply regexp substitution.
+##
+##   - Indent(chars="  "): will indent the text with the prefix
+##     Please remember that template engines gets also to modify the text and
```
Comment (GodenYao) [inline .gitchangelog.rc]: ...engines get?

**Comment by vrajat** [comment]
This branch has too many conflicts. I;ve opened #150

**Pull request closed without merging**